pkg/api/vlabs/validate.go

```diff
@@ -547,6 +547,21 @@ func (a *Properties) validateAddons() error {
 					if IsNSeriesSKU && sv.LT(minVersion) {
 						return errors.New("NVIDIA Device Plugin add-on can only be used Kubernetes 1.10 or above. Please specify \"orchestratorRelease\": \"1.10\"")
 					}
+					if a.HasCoreOS() {
```

Comment:
Let's add some validation unit tests. We can either add to `Test_Properties_ValidateAddons` or create another test following that pattern to verify that coreos + these addons will throw the expected error.

Reply:
Added some minimal validation that these addons won't be accepted with coreos.